pkg/resource/deploy/step_generator.go

```diff
@@ -333,175 +334,224 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, res
 	//  In this case, the resource we are operating upon now exists in the old snapshot.
 	//  It must be an update or a replace. Which operation we do depends on the the specific change made to the
 	//  resource's properties:
+	//
+	//  - if the user has requested that only specific resources be updated, and this resource is
+	//    not in that set, do no 'Diff' and just treat the resource as 'same' (i.e. unchanged).
+	//
 	//  - If the resource's provider reference changed, the resource must be replaced. This behavior is founded upon
 	//    the assumption that providers are recreated iff their configuration changed in such a way that they are no
 	//    longer able to manage existing resources.
+	//
 	//  - Otherwise, we invoke the resource's provider's `Diff` method. If this method indicates that the resource must
 	//    be replaced, we do so. If it does not, we update the resource in place.
 	if hasOld {
 		contract.Assert(old != nil)
-		diff, err := sg.diff(urn, old, new, oldInputs, oldOutputs, inputs, prov, allowUnknowns, goal.IgnoreChanges)
-		if err != nil {
-			// If the plugin indicated that the diff is unavailable, assume that the resource will be updated and
-			// report the message contained in the error.
-			if _, ok := err.(plugin.DiffUnavailableError); ok {
-				diff = plugin.DiffResult{Changes: plugin.DiffSome}
-				sg.plan.ctx.Diag.Warningf(diag.RawMessage(urn, err.Error()))
-			} else {
-				return nil, result.FromError(err)
+
+		// If the user requested only specific resources to update, and this resource was not in
+		// that set, then do nothin but create a SameStep for it.
+		if updateTargetsOpt != nil && !updateTargetsOpt[urn] {
+			logging.V(7).Infof(
+				"Planner decided not to update '%v' due to not being in target group (same) (inputs=%v)", urn, new.Inputs)
+		} else {
+			updateSteps, res := sg.generateStepsFromDiff(
+				event, urn, old, new, oldInputs, oldOutputs, inputs, prov, goal)
+
+			if res != nil {
```

Comment:
Nit - the vertical whitespace here stands out as inconsistent with the style of the rest of this file.  In particular, the empty line between the call and its error check is unusual for our Go codebase.

Reply:
the primary style i take is one where things feel comfortable.  When the previous line has to wrap, as above, it already feels cramped and cluttered to me as somethin large to wrap my mind around.  Having the next statement abut it just adds visual clutter for me and makes it harder for me to 'pace out' the statements.

It's similar, in my mind, to how sentences vs. paragraphs work in normal writing.  

--

Regardless, i find that these style questions *really* slow down PRs.  I'm already 100% on board with the linter stopping things and having to fit that style.  Having to rigidly conform on a bunch of *unstated* things is more difficult to deal with and ends up taking a lot of time to deal with.  i.e. if i chance this i have ot wait another hour, babysit hte PR and whatnot.  I'd really prefer we be able to find a reasonable balance here.  I'm trying to keep the code readable and clear, without gross deviation.  But i'm likely never going to exactly match how everyone else has their personal preferences on things.

Note, my ideal here would be: if this is something we want to codify, we just do so with a tool.  Then there's no guesswork.  I can simply run the tool and get the output that is desired.  That can be done preemptively, not in the middle of the PR.